### PR TITLE
Merge compose branch into main

### PR DESCRIPTION
**Pull request by arriolac**
Merges `compose` branch into `main`.

## Changes
- Merge main into compose.
- Update Compose in RecyclerView implementation. (#766)
- [Compose]: Implement RecyclerView item in Compose. (#763)
- [Compose]: Update dependencies and remove reference to accompanist (#760)
- [Compose]: Upgrade to 1.2.0-alpha07 (#758)
- Update tooling dependencies (#754)
- [Compose] Update to rc02 (#731)
- [Compose] Update to beta07 (#716)
- Update to Compose beta02 (#708)
- @manuelvicnt Update to Compose beta01 (#704)
- Updates to Compose alpha12 (#697)
- Updates Compose to alpha11 (#693)
- Updates to Compose alpha10 (#683)
- Update to alpha08 version of Compose (#677)
- Updates Compose to alpha07 (#670)
- Updates Compose to alpha06 (#666)
- Updates to compose alpha05 (#664)
- Updates to Compose alpha04 (#657)
- Updates to Compose alpha (#639)
- Migrates Plant details screen to Compose (#1) (#638)

### CODE REVIEW

File: app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt [hunk 1]
```diff
@@ -49,68 +44,23 @@ class PlantDetailFragment : Fragment() {
...
-            toolbar.setNavigationOnClickListener { view ->
-                view.findNavController().navigateUp()
-            }
-
-            toolbar.setOnMenuItemClickListener { item ->
-                when (item.itemId) {
-                    R.id.action_share -> {
+    ) = ComposeView(requireContext()).apply {
+
+        setContent {
```
Comment (bentrengrove) [inline app/src/main/java/com/google/samples/apps/sunflower/PlantDetailFragment.kt:49]: Do we still have to set the lifecycle for compose?

File: app/src/main/java/com/google/samples/apps/sunflower/compose/utils/TextUtils.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ * TODO: Remove when https://issuetracker.google.com/issues/158065051 is fixed
+ *
+ * @param id the resource identifier
+ * @param quantity The number used to get the correct string for the current language's
+ *           plural rules.
+ *
+ * @return the string data associated with the resource
+ */
+@Composable
+fun getQuantityString(@PluralsRes id: Int, quantity: Int): String {
```
Comment (bentrengrove) [inline app/src/main/java/com/google/samples/apps/sunflower/compose/utils/TextUtils.kt:36]: This exists in Compose now I think, https://developer.android.com/jetpack/compose/resources#string-plurals

Comment (arriolac) [reply app/src/main/java/com/google/samples/apps/sunflower/compose/utils/TextUtils.kt:36]: This does exactly that. Looks like the FR for this function to be a part of Compose is still open https://issuetracker.google.com/issues/171516623

Comment (bentrengrove) [reply app/src/main/java/com/google/samples/apps/sunflower/compose/utils/TextUtils.kt:36]: Doh, docs are out of date. Its in 1.2.0-alpha06

Comment (arriolac) [reply app/src/main/java/com/google/samples/apps/sunflower/compose/utils/TextUtils.kt]: Fixed in 1fa90f4

**Review by tunjid — APPROVED** [review]
LGTM!

**Comment by Giangpro89** [comment]
Cam on

**Pull request merged**